OTE_SDK. Increase code coverage. entities/metrics.py, entities/graph.py

saltykox opened the pull request:
Test list:
- ote_sdk/ote_sdk/tests/entities/test_metrics.py
- ote_sdk/ote_sdk/tests/entities/test_graph.py
- ote_sdk/ote_sdk/tests/entities/test_id.py
- ote_sdk/ote_sdk/tests/entities/interfaces/test_graph_interface.py
- ote_sdk/ote_sdk/tests/configuration/enums/test_model_lifecycle.py
- ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py

Task: CVS-72106

saltykox commented:
run ote_sdk tests

saltykox commented:
@LeonidBeynenson please review

saltykox commented:
run ote_sdk tests

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/configuration/enums/test_model_lifecycle.py:
```diff
@@ -0,0 +1,46 @@
+        Check ModelLifecycle Enum class elements
+
+        <b>Expected results:</b>
+        Test passes if ModelLifecycle Enum class length equal expected value and its elements have expected
```
```suggestion
        Test passes if ModelLifecycle Enum class length is equal to expected value and its elements have expected
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/interfaces/test_graph_interface.py:
```diff
@@ -0,0 +1,64 @@
+    def test_i_graph(self):
+        """
+        <b>Description:</b>
+        Check IGraph class object initiation
```
```suggestion
        Check IGraph class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_id.py:
```diff
@@ -0,0 +1,60 @@
+    def test_id(self):
+        """
+        <b>Description:</b>
+        Check ID class object initiation
```
```suggestion
        Check ID class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_evaluate_interface(self):
+        """
+        <b>Description:</b>
+        Check IEvaluationTask class object initiation
```
```suggestion
        Check IEvaluationTask class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_i_inference_task(self):
+        """
+        <b>Description:</b>
+        Check IInferenceTask class object initiation
```
```suggestion
        Check IInferenceTask class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_i_raw_inference(self):
+        """
+        <b>Description:</b>
+        Check TestIRawInference class object initiation
```
```suggestion
        Check TestIRawInference class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_optimization_interface(self):
+        """
+        <b>Description:</b>
+        Check IOptimizationTask class object initiation
```
```suggestion
        Check IOptimizationTask class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_training_interface(self):
+        """
+        <b>Description:</b>
+        Check ITrainingTask class object initiation
```
```suggestion
        Check ITrainingTask class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_unload_interface(self):
+        """
+        <b>Description:</b>
+        Check IUnload class object initiation
```
```suggestion
        Check IUnload class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+        Check ExportType Enum class elements
+
+        <b>Expected results:</b>
+        Test passes if ExportType Enum class length equal expected value and its elements have expected
```
```suggestion
        Test passes if ExportType Enum class length is equal to expected value and its elements have expected
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+    def test_export_interface(self):
+        """
+        <b>Description:</b>
+        Check IExportTask class object initiation
```
```suggestion
        Check IExportTask class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/usecases/tasks/interfaces/test_interfaces.py:
```diff
@@ -0,0 +1,297 @@
+        Check OptimizationType Enum class elements
+
+        <b>Expected results:</b>
+        Test passes if OptimizationType Enum class length equal expected value and its elements have expected
```
```suggestion
        Test passes if OptimizationType Enum class length is equal to expected value and its elements have expected
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_graph.py:
```diff
@@ -0,0 +1,981 @@
+    def compare_list_elements(actual_list: list, expected_list: list) -> None:
+        """Function to compare lists without checking elements sequence"""
+        assert len(actual_list) == len(expected_list)
+        for actual_element in actual_list:
```
actual_list      = [1, 1, 1, 2, 3]
expected_list = [1, 2, 3, 4, 5]

compare_list_elements(actual_list, expected_list) - > ?

saltykox replied:
This function performs assertion checks, and not returns bool values

Ilya-Krylov replied:
is it ok that set(actual_list) is subser of set(expected_list) but not vice versa in general case?

saltykox replied:
Thanks for the comment, I think it's better to use a direct comparison, I will fix incorrect comparison

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_graph.py:
```diff
@@ -0,0 +1,981 @@
+    def test_graph_initiation(self):
+        """
+        <b>Description:</b>
+        Check Graph class object initiation
```
```suggestion
        Check Graph class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_graph.py:
```diff
@@ -0,0 +1,981 @@
+        method return expected values
+
+        <b>Steps</b>
+        1. Check initiation of non-directed Graph object
```
```suggestion
        1. Check initialization of non-directed Graph object
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_graph.py:
```diff
@@ -0,0 +1,981 @@
+
+        <b>Steps</b>
+        1. Check initiation of non-directed Graph object
+        2. Check initiation of directed Graph object
```
```suggestion
        2. Check initialization of directed Graph object
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_graph.py:
```diff
@@ -0,0 +1,981 @@
+    def test_multi_di_graph(self):
+        """
+        <b>Description:</b>
+        Check MultiDiGraph class object initiation
```
```suggestion
        Check MultiDiGraph class object initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_metrics.py:
```diff
@@ -104,3 +178,918 @@ def test_matrix_metric(self):
+
+        <b>Steps</b>
+        1. Check name, ys, xs and type attributes of CurveMetric object
+        2. Check name, ys, xs and type attributes of CurveMetric object with not specified xs initiation parameter
```
```suggestion
        2. Check name, ys, xs and type attributes of CurveMetric object with not specified xs initialization parameter
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_metrics.py:
```diff
@@ -104,3 +178,918 @@ def test_matrix_metric(self):
+        <b>Steps</b>
+        1. Check name, ys, xs and type attributes of CurveMetric object
+        2. Check name, ys, xs and type attributes of CurveMetric object with not specified xs initiation parameter
+        3. Check ValueError exception raised when length of ys CurveMetric object initiation parameter not equal length
```
```suggestion
        3. Check ValueError exception raised when length of ys CurveMetric object initialization parameter is not equal to length
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_metrics.py:
```diff
@@ -104,3 +178,918 @@ def test_matrix_metric(self):
+        palette and visualisation_type parameters
+        2. Check name, palette and type attributes and __repr__ method values for BarChartInfo object with not
+        specified palette and visualisation_type parameters
+        3. Check ValueError exception raised when visualization_type BarChartInfo object initiation parameter not equal
```
```suggestion
        3. Check ValueError exception raised when visualization_type BarChartInfo object initialization parameter is not equal
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_metrics.py:
```diff
@@ -72,18 +140,24 @@ def test_matrix_metric(self):
         1. Create Matrices
         2. Check normalized matrices against pre-computed matrices
+        3. Check positive scenario when row_labels and column_labels parameters specified during MatrixMetric object
+        initiation
```
```suggestion
        initialization
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_metrics.py:
```diff
@@ -72,18 +140,24 @@ def test_matrix_metric(self):
         2. Check normalized matrices against pre-computed matrices
+        3. Check positive scenario when row_labels and column_labels parameters specified during MatrixMetric object
+        initiation
+        4. Check ValueError exception raised when row_labels parameter length not equal to number of rows of
```
```suggestion
        4. Check ValueError exception raised when row_labels parameter length is not equal to number of rows of
```

Ilya-Krylov commented on ote_sdk/ote_sdk/tests/entities/test_metrics.py:
```diff
@@ -72,18 +140,24 @@ def test_matrix_metric(self):
+        initiation
+        4. Check ValueError exception raised when row_labels parameter length not equal to number of rows of
+        MatrixMetric object
+        5. Check ValueError exception raised when column_labels parameter length not equal to number of columns of
```
```suggestion
        5. Check ValueError exception raised when column_labels parameter length is not equal to number of columns of
```

saltykox replied:
Suggestions applied, thanks

Ilya-Krylov commented:
run ote_sdk tests

Ilya-Krylov commented:
Could you double check that warnings related ote_sdk testing are valid and suppress them one-by-one ?

See http://213.221.44.203/job/ote/job/precommit/job/ote_sdk_tests_runner/191/console

```
17:12:35  entities/test_graph.py: 13 warnings
17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/venv/lib/python3.8/site-packages/numpy/matrixlib/defmatrix.py:69: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
17:12:35      return matrix(data, dtype=dtype, copy=False)
17:12:35  
17:12:35  entities/test_metrics.py::TestMatrixMetricsGroup::test_matrix_metrics_group
17:12:35  entities/test_metrics.py::TestPerformance::test_performance
17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/ote_sdk/entities/metrics.py:371: RuntimeWarning: invalid value encountered in true_divide
17:12:35      self.__matrix_values = self.__matrix_values.astype(
```

saltykox commented:
run ote_sdk tests

Ilya-Krylov commented:
As far as I can see not all suggestions have been commited AND please respond here https://github.com/openvinotoolkit/training_extensions/pull/750#discussion_r766584010

Ilya-Krylov commented:
> Could you double check that warnings related ote_sdk testing are valid and suppress them one-by-one ?
> 
> See http://213.221.44.203/job/ote/job/precommit/job/ote_sdk_tests_runner/191/console
> 
> ```
> 17:12:35  entities/test_graph.py: 13 warnings
> 17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/venv/lib/python3.8/site-packages/numpy/matrixlib/defmatrix.py:69: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
> 17:12:35      return matrix(data, dtype=dtype, copy=False)
> 17:12:35  
> 17:12:35  entities/test_metrics.py::TestMatrixMetricsGroup::test_matrix_metrics_group
> 17:12:35  entities/test_metrics.py::TestPerformance::test_performance
> 17:12:35    /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/ote_sdk/entities/metrics.py:371: RuntimeWarning: invalid value encountered in true_divide
> 17:12:35      self.__matrix_values = self.__matrix_values.astype(
> ```

dont suppress openvino related import

saltykox commented:
- fixed check_graph_list_attributes method, changed the way to check graph edges and nodes
- added suppression of "PendingDeprecationWarning" in test_graph.py and suppression of "invalid value encountered in true_divide" in test_metrics.py

Ilya-Krylov commented:
run ote_sdk tests

saltykox commented:
run ote_sdk tests

saltykox commented:
run ote_sdk tests
